### PR TITLE
React Capstone Project (PR)

### PR DESCRIPTION
**Pull request by teke85**
# The following project Requirements  were fulfilled

- Build a single-page application (SPA) with React and Redux.
- Retrieve data from the API should be stored in the Redux store.
- Filtered the data that retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) have a unique route within the SPA.
- The project deployed and accessible online (add a link to the online version of your app to your README file).
screencapture-localhost-3000-2023-02-15-20_19_28

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,248 @@
...
+npm install history
+npm install --save-dev jest
+npm install --save-dev babel-jest @babel/core @babel/preset-env
+npm install --save-dev jest @testing-library/react
+npm install --save-dev redux react-redux
+npm install --save-dev jest-environment-jsdom
+npm install --save-dev @testing-library/jest-dom
+npm i redux-mock-store
+```
+ -->
```
Comment (zieeco) [inline README.md]: - Let's remove this commented code to keep our readme clean.

File: README.md [hunk 2]
```diff
@@ -1 +1,248 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE.txt) licensed.
+
+_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
+
```
Comment (zieeco) [inline README.md]: - Please, kindly consider removing this placeholder text as the text was added to guide you on how to add content to it appropriate section.

File: README.md [hunk 3]
```diff
@@ -1 +1,254 @@
...
+To run tests, run the following command:
+
+```sh
+  npm test
+```
+
+### Deployment
+
+You can deploy this project using:
```
Comment (zieeco) [inline README.md:170]: - Since the app is deployed, kindly state which platform you've used in deploying the app.

File: src/pages/Countries.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import { getCountries } from '../Redux/Country/countriesSlice';
+import Country from '../components/Country';
+import Wrapper from '../components/Wrapper';
+
+const Countries = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const continent = JSON.parse(searchParams.get('continent') || '{}');
```
Comment (zieeco) [inline src/pages/Countries.js:16]: - Hey, we should be able to search for countries by its name by adding a search bar where users can search for countries by their names. this should be implemented using the filter method. kindly add this feature. see the image below on where to add it.

*search requirement*
Screenshot from 2023-02-17 23-32-14



Screenshot from 2023-02-17 23-25-19

File: src/_test_/App.test.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  render(
+    <Provider store={store}>
+      <BrowserRouter>
+        <App />
+      </BrowserRouter>
+    </Provider>,
+  );
+  const container = await screen.findByTestId('app-container');
+  expect(container).toBeInTheDocument();
+});
```
Comment (zieeco) [inline src/_test_/App.test.js]: - Your test related to the app is falling. please, kindly fix this. see the image below:


Screenshot from 2023-02-17 23-43-31

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @teke85 ,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Professional readme file.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (teke85) [reply README.md]: Hello @zieeco thank you for your comment. I have removed the commented code.

File: README.md (hunk 2 above)
Comment (teke85) [reply README.md]: Hello @zieeco . place holder has been removed thanks.

File: README.md (hunk 3 above)
Comment (teke85) [reply README.md:170]: Hi @zieeco I have done as suggested.

File: src/pages/Countries.js (hunk 4 above)
Comment (teke85) [reply src/pages/Countries.js:16]: Hi @zieeco This feature has been implemented.

React App (3)

File: src/_test_/App.test.js (hunk 5 above)
Comment (teke85) [reply src/_test_/App.test.js]: Hi @zieeco Thank you for your comment. I have resolved this error.

**Review by BenMukebo — APPROVED** [review]
# Status: Approved  :champagne:  ✔️:confetti_ball: 

## Hi @teke85, 

### **Congratulations! 🎉 Project Approved ✅**

Good job so far. Well Done.👍🏻 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight 🎉
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 
- The  filter has been weel implemented. ✔️ 
- All the tests are passing. ✔️ 
- Professional README file. :heavy_check_mark:
- Your app has been well deployed heavy_check_mark
- Your Design looks awesome, I like your Animation. 🥇  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**